src/main/java/com/nirmata/workflow/queue/zookeeper/SimpleQueue.java

```diff
@@ -331,4 +331,18 @@ private static long getEpoch(String itemNode)
         }
         return 0;
     }
+    
+    @Override
+	public void closeGraceFully(long timeOut, TimeUnit timeUnit) {
+		 if ( started.compareAndSet(true, false) )
+	        {
+	            executorService.shutdown();
+	            try {
+					executorService.awaitTermination(timeOut, timeUnit);
+				} catch (InterruptedException e) {
+					log.error("Exception occurred while waiting for running tasks to complete", e);
```

Comment:
Please add a call to Thread.currentThread().interrupt() (see: http://www.javapractices.com/topic/TopicAction.do?Id=251)